### PR TITLE
Company

### PR DESCRIPTION
**Pull request by ctlong**
This is a precursor to having the mobile app login as just entering the company code.

### CODE REVIEW

File: app/controllers/companies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
+class CompaniesController < ApplicationController
+  def create
+    code = String.random_alphanumeric(8) 
+    while Company.where(code: code).length != 0
```
Comment (h8rry) [inline app/controllers/companies_controller.rb:4]: `.length != 0` -> `.any?`

Comment (ggharry) [inline app/controllers/companies_controller.rb]: I don't understand the purpose of this while loop

Comment (h8rry) [inline app/controllers/companies_controller.rb:4]: You are trying to generate a unique code. It's generally a very bad experience to join a company just based on some code.

File: app/controllers/companies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      code = String.random_alphanumeric(8)
+    end
+    @company = Company.create(name: params[:company][:name], code: code)
+  end
+
+  def show
+    @company = Company.find(params[:id])
+  end
+
+  def update
```
Comment (h8rry) [inline app/controllers/companies_controller.rb:14]: Do you need this update method?

File: app/controllers/companies_controller.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      if @company.destroy
+        render json: {deleted: true}
+      else
+        render json: {deleted: false}, status: :bad_request
+      end
+    end
+  end
+
+  def match
+    @company = Company.where(code: params[:code])[0]
```
Comment (h8rry) [inline app/controllers/companies_controller.rb:45]: `@company = Company.finy_by(code: params[:code])`

**Pull request closed without merging**